packages/fxa-auth-server/lib/routes/attached-clients.js

```diff
@@ -227,19 +227,7 @@ module.exports = (log, db, devices, clientUtils) => {
           payload: 'required',
         },
         validate: {
-          payload: isA
-            .object({
-              clientId: validators.clientId.allow(null).optional(),
-              sessionTokenId: isA
-                .string()
-                .regex(HEX_STRING)
-                .allow(null)
-                .optional(),
-              refreshTokenId: validators.refreshToken.allow(null).optional(),
-              deviceId: DEVICES_SCHEMA.id.allow(null).optional(),
-            })
-            .or('clientId', 'sessionTokenId', 'refreshTokenId', 'deviceId')
-            .with('refreshTokenId', ['clientId']),
+          payload: validators.destroyDevice,
```

Comment:
I like that you're thinking of how to test the validation. I also like having the validation defined inline because it's easier to reference when you're working on the route. We don't need to find a solution in this PR since the bug is unrelated but I'd prefer the test to do the flexing in order to keep the code more readable than the other way around. Let's leave it out of this PR though.

Reply:
No problem, I was also torn on this as well. I've gotten thrown off a couple times because the unit tests on the handlers bypass the validators. In this case, the validators for this endpoint seem to be untested, so I made this change in an effort to help pinpoint the bug. I've rolled back the change. I do agree it's nice to see the validation rules inline.